### PR TITLE
acme_diags dependency

### PR DESCRIPTION
**Pull request by zshaheen**
* Added dependencies for `acme_diags`
* Also dumped a working environment into `acme_unified.yml`
* Removed dependencies not explicitly needed, in hopes of reducing the file count of the environment.

### CODE REVIEW

**Comment by zshaheen** [comment]
@doutriaux1 @xylar Once you guys give the okay and it's merged to master, I'll update the confluence page with the new versions of the software.

File: acme_unified.yml [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+  - jupyter-client==5.1.0
+  - jupyter-console==5.2.0
+  - jupyter-core==4.4.0
+  - mv2==3.0.0
+  - nb-conda==2.2.1
+  - nb-conda-kernels==2.1.0
+  - output-viewer==1.2.2
+  - prompt-toolkit==1.0.15
+  - regrid2==3.0.0
+prefix: /export_2/shaheen2/anaconda2/envs/acme_unified_env-might-work-charles2
```
Comment (xylar) [inline acme_unified.yml]: Can we remove this before the PR gets merged?

File: acme_metapackage.bash [hunk 2]
```diff
@@ -46,8 +29,8 @@ conda metapackage -c conda-forge -c acme -c uvcdat -c opengeostat acme-unified $
 "cartopy  ${OPERATOR}0.15.1" \
 "progressbar ${OPERATOR}2.3" \
 "scipy <1.0.0" \
-"matplotlib" \
-"basemap" \
```
Comment (xylar) [inline acme_metapackage.bash]: matplotlib and basemap get added via acme_diags, I assume?

**Comment by xylar** [comment]
@zshaheen, this is great!  Could you also add a commit to update dask to 0.15.4, nco to 4.7.0, lxml to 4.1.1, netcdf4 to 1.3.1, shapely to 1.6.2 and add pillow 4.3.0?  That would be really helpful.  Then, I think we would be all set for 1.1.2.

Once you've done that, you'll probably want to test that acme_diags still works with the new versions of those various packages.  If you create a new yaml file, I'll run a quick test or 2 on MPAS-Analysis to make sure it's unaffected by the changes as well.

Thanks!

File: acme_unified.yml (hunk 1 above)
Comment (zshaheen) [reply acme_unified.yml]: I'm guessing you're talking about the `prefix` (line 230) right? That might break something. In our other yml's, we keep this prefix and it works fine across many machines with different users. But I will change the prefix to `/export_2/shaheen2/anaconda2/envs/acme_unified_env`.

I also need to remove the build numbers as well.

File: acme_metapackage.bash (hunk 2 above)
Comment (zshaheen) [reply acme_metapackage.bash]: Well, `matplotlib` is a dependency of `acme_diags`. As for `basemap`, why did we remove that @doutriaux1?

Comment (xylar) [reply acme_metapackage.bash]: We definitely need both matplotlib and basemap for MPAS-Analysis.  Is there any harm in leaving them in here?

File: acme_unified.yml (hunk 1 above)
Comment (xylar) [reply acme_unified.yml]: Hmm, I really don't understand how the prefix can be useful but if you feel like it's needed you can change it as you suggest.

Comment (xylar) [reply acme_unified.yml]: https://stackoverflow.com/questions/41274007/anaconda-export-environment-file
According to this post, the prefix gets ignored and can be removed.

File: acme_metapackage.bash (hunk 2 above)
Comment (doutriaux1) [reply acme_metapackage.bash]: it won't hurt anything, since they're already brought in by acme_diags.

File: acme_unified.yml (hunk 1 above)
Comment (doutriaux1) [reply acme_unified.yml]: @xylar @zshaheen yes I would hope it's ignored.

**Comment by zshaheen** [comment]
@xylar When I update `nco` to `4.7.0`, I get the following error:
```
WARNING:conda_build.environ:failed to get install actions, retrying.  exception was: The following specifications were found to be in conflict:
  - nco ==4.7.0
  - vcs ==2.12
Use "conda info <package>" to see the dependencies for each package.
```

Everything is fine when `nco` is `4.6.9`.

Here's the conda info for both `nco` `4.7.0` and `vcs` `2.12`:

```
(acme_unified_env-might-work-charles2) shaheen2@aims4:~/github/conda-recipes$ conda info vcs==2.12
Fetching package metadata ...................

vcs 2.12 py27had75802_0
-----------------------
file name   : vcs-2.12-py27had75802_0.tar.bz2
name        : vcs
version     : 2.12
build string: py27had75802_0
build number: 0
channel     : uvcdat
size        : 1.2 MB
arch        : x86_64
has_prefix  : True
md5         : c1e5af3eb5908648314ecff3e20de488
noarch      : None
platform    : linux
requires    : ()
subdir      : linux-64
timestamp   : 1504625128494
url         : https://conda.anaconda.org/uvcdat/linux-64/vcs-2.12-py27had75802_0.tar.bz2
dependencies:
    cdat_info
    cdms2
    cdutil
    dv3d
    genutil
    numpy
    python >=2.7,<2.8.0a0
    requests
    vtk-cdat >7.1.0.2.8
(acme_unified_env-might-work-charles2) shaheen2@aims4:~/github/conda-recipes$ conda info nco=4.7.0
Fetching package metadata ...................

nco 4.7.0 0
-----------
file name   : nco-4.7.0-0.tar.bz2
name        : nco
version     : 4.7.0
build string: 0
build number: 0
channel     : conda-forge
size        : 6.4 MB
arch        : x86_64
has_prefix  : True
license     : GPL-3.0
md5         : 6f0f9831f70e581a4dff97f218f91610
noarch      : None
platform    : linux
requires    : ()
subdir      : linux-64
url         : https://conda.anaconda.org/conda-forge/linux-64/nco-4.7.0-0.tar.bz2
dependencies:
    esmf
    expat 2.1.*
    gsl >=2.2,<2.3
    hdf5 1.10.1
    libgcc
    libnetcdf 4.4.*
    openblas 0.2.19|0.2.19.*
    udunits2
```

Any idea's @doutriaux1? We might want to ask Charlie Zender what new dependencies or specific version requirements were added to `4.7.0` of `nco`.

**Comment by xylar** [comment]
@zshaheen, thanks for testing.  I don't see an obvious conflict there.  It might be in dependencies of dependencies.  I didn't change any dependencies in the nco recipe when I pushed the PR to support 4.7.0 on conda-forge.  I believe that build 1 of 4.6.9 did change at least one dependency.  I'll track it down and post below.

**Comment by xylar** [comment]
@zshaheen and @doutriaux1, the following PR (for nco 4.6.9 build 3, not build 1) updated to hdf5 1.10.1, which seems like the most likely source of conflict:
https://github.com/conda-forge/nco-feedstock/pull/46
Could it be that any of the dependencies of vcs depend on hdf5?

Do you know which build of nco 4.6.9 you are testing?

**Comment by xylar** [comment]
@zshaheen, could you maybe let me know what you did or give me a yaml file to try out or something?  That way I could try to reproduce the error and hopefully help find a solution.

**Comment by zshaheen** [comment]
@xylar If you try running `acme_metapackage.bash` under the `acme_diags` branch, you should see the error.

**Comment by xylar** [comment]
@zshaheen, I'm just getting back to this.  I had to update my conda-build to support the -c option (which requires a version not yet available on conda-forge).  I can reproduce your error and I'm looking into it.

**Comment by xylar** [comment]
@zshaheen and @doutriaux1, after significant digging around, I seem to have located at least one conflict: hdf5 1.10.1 is used by nco but 1.8.18 is required by vtk-cdat.  1.10.1 seems to be the "official" pinned version on conda-forge:
https://github.com/conda-forge/conda-forge.github.io/blob/master/scripts/pin_the_slow_way.py#L56
Is there a way for you guys to make a build of uv-cdat with this version of hdf5?  Otherwise, we're going to hit a snag with e3sm-unified.

**Comment by xylar** [comment]
expat and esmf also have different versions in nco than in uv-cdat but the metapackage builds fine if I include the nco versions explicitly so I'm not sure these packages are being pinned anywhere within uv-cdat.  But something to keep an eye on.

**Comment by xylar** [comment]
libnetcdf also seems like a potential place for conflict.  The "slow-pinned" version, also used by nco, is 4.4.*, while something in uv-cdat seems to require 4.3.3.1.  To make matters worse, netcdf4 seems to use 4.5.*

**Comment by xylar** [comment]
Ignore my earlier comments about expat, esmf and libnetcdf.  It seems like the pinned versions of those libraries are compatible with nco.  So the problem is almost definitely hdf5.  I think a rebuild of vtk-cdat that includes 1.10.1 would probably fix the problem (assuming no incompatibilities).  I'll leave it up to you if you want to add 1.10.1 or replace 1.8.18 with 1.10.1.

**Comment by doutriaux1** [comment]
all the cdms stack is built against hdf 1.8 it will probably not matter except that hdf5 will not run unless you set some env variable (because compiled against a diff version hdf5)
@czender any chance you could build . a version built against hdf5 1.8.* and libnetcdf 4.4.*

**Comment by doutriaux1** [comment]
we will eventually (i.e coming month) nuild nighlties against hdf5 1.10 but there's a huge list of software to port. Conda-forge started to rebuild package against 1.10 slowly

**Comment by doutriaux1** [comment]
We can also build our own local version of nco (same version) against the correct hdf5 and host it on acme channel

**Comment by xylar** [comment]
@czender is not the one who maintains NCO on conda-forge.  I could modify the recipe to add 1.8.18 back in.  I'll check with the other maintainers to make sure they don't think that's a bad idea.

**Comment by czender** [comment]
NCO will work with any version of HDF5 that netCDF4 is built with. However, conda-forge depends on HDF5 for many packages so presumably the conda-forge NCO maintainer who updated NCO from HDF5 1.8 to 1.10 did so in coordination with those other packages. @xylar knows who to ask, and I'll cross my fingers the response does not cause anyone a lot of work.

**Comment by xylar** [comment]
@doutriaux1, it doesn't seem like it's all that common to have multiple versions of a given package version with different dependencies on conda-forge.  I'm looking into how to do it but it's clearly not as simple as specifying multiple dependencies.  (This can be done if, for example, one package version is used in linux and another in windows or similarly for python 2x vs. 3x, but that's a different case than what we're trying to address.)

**Comment by xylar** [comment]
@doutriaux1, I really don't like the idea of having a separate build of nco on the uv-cdat channel but that may be our only good option for having a separate build with hdf5 1.8.18.

**Comment by doutriaux1** [comment]
@xylar my bad, I meant the acme/e3sm channel, typed too fast. Again conda-forge is moving all the packages to hdf5 1.10 so we will be there in a few weeks. Another way is to force hdf5 1.8 or 1.10 I'm 99% confident it won't affect us. BUT by default hdf5 refuse to work if it running with a version differne tfrom the one it was compiled against and therefore you must set an env variable to allow for this. (I can't remember the exact env variable, but the hdf5 error message indicates it)

**Comment by xylar** [comment]
Okay, so I would suggest just this once that we put nco onto the e3sm channel and use it for acme-unified.  Then, hopefully the next update of (what will then be renamed) e3sm-unified will all be with 1.10.* and we can use the version of nco from conda-forge.  Does that sound like a plan?

**Comment by xylar** [comment]
@zshaheen, this is *super* old and can probably be closed, right?

**Comment by zshaheen** [comment]
@xylar, yeah this is pretty old, lets close this.

**Pull request closed without merging**